lyse/routines.py

```diff
@@ -0,0 +1,557 @@
+#####################################################################
+#                                                                   #
+# /routines.py                                                      #
+#                                                                   #
+# Copyright 2013, Monash University                                 #
+#                                                                   #
+# This file is part of the program lyse, in the labscript suite     #
+# (see http://labscriptsuite.org), and is licensed under the        #
+# Simplified BSD License. See the license.txt file in the root of   #
+# the project for the full license.                                 #
+#                                                                   #
+#####################################################################
+
+"""
+analysis GUI code
+"""
+
+import os
+import time
+import logging
+import threading
+import subprocess
+
+# Labscript imports
+from labscript_utils.qtwidgets.headerview_with_widgets import HorizontalHeaderViewWithWidgets
+
+# qt imports
+from qtutils.qt import QtCore, QtGui, QtWidgets
+from qtutils import inmain_decorator, UiLoader, DisconnectContextManager
+
+from lyse import LYSE_DIR
+import lyse.widgets
+import lyse.routines
```

Comment:
Removed.  It is interesting that the code worked with that there since it is self referential.

Reply:
Since it looks like this package of changes is pretty much tied up, I will move on to worker sub-processes and windows.  In the short term I will introduce a *new style* worker that will be dispatched if the worker script starts with a comment something like 
```python
# analysis_version = "2.0"   
```
this should allow easy backward compatibility.